source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -221,12 +221,23 @@ const std::string& OAuth2Filter::bearerPrefix() const {
 
 /**
  * primary cases:
- * 1) user is signing out
- * 2) /_oauth redirect
- * 3) user is authorized
- * 4) user is unauthorized
+ * 1) pass through header is present
+ * 2) user is signing out
+ * 3) /_oauth redirect
+ * 4) user is authorized
+ * 5) user is unauthorized
  */
 Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+  // Stop Filter if Passthrough header is present
```

Comment:
I wouldn't use the term Stop here since it makes it sound like we're going into Stop iteration

Maybe something like `// Do nothing if this matches one of the pass through matchers`?